pandas/io/gbq.py

```diff
@@ -148,6 +151,43 @@ def get_credentials(self):
 
         return credentials
 
+    def get_credentials_for_service_account(self):
```

Comment:
but you are fragmenting code. so pls follow my instructions above (either a sep function for both or inline them both)